nexus/src/app/sagas/region_replacement_start.rs

```diff
@@ -554,7 +555,30 @@ async fn srrs_replace_region_in_volume(
         .await
         .map_err(ActionError::action_failed)?;
 
-    Ok(())
+    info!(log, "replacement returned {:?}", volume_replace_region_result);
+
+    match volume_replace_region_result {
+        VolumeReplaceResult::AlreadyHappened | VolumeReplaceResult::Done => {
+            // The replacement was done either by this run of this saga node, or
+            // a previous one (and this is a rerun). This can only be returned
+            // if the transaction occurred on the non-deleted volume so proceed
+            // with the rest of the saga (to properly clean up allocated
+            // resources).
+
+            Ok(())
+        }
+
+        VolumeReplaceResult::ExistingVolumeDeleted => {
+            // Unwind the saga here to clean up the resources allocated during
+            // this saga. The associated background task will transition this
+            // request's state to Completed.
+
+            Err(ActionError::action_failed(format!(
+                "existing volume {} deleted",
+                old_volume_id
+            )))
```

Comment:
should this be `Error::conflict`? i suppose it doesn't really matter since these sagas are triggered by a background task, right?

Reply:
even so, it should be a conflict, yeah: done in d2135f21b